### ISSUE TITLE
After installing shadow in chroot_buildroot_*, overriding PATH is broken for pmos user

### ISSUE THREAD

**Issue by ollieparanoid** [postmarketOS/pmbootstrap#1071]
Thanks to @drebrez for reporting in #postmarketOS:
> I found out a possible issue, in `postmarketos-base` we install `shadow` which changes the PATH env variable of the pmos user, `postmarketos-base` is only installed in the device chroot except when you try to build the package itself, which also install the dependencies and `shadow`. All the subsequent commands executed as `su pmos -c '....'` are broken 

```
drebrez@drebrez-laptop:~/pmbootstrap$ pmbootstrap chroot -b
[02:21:55] (buildroot_aarch64) % sh
[02:21:55] (native) install alpine-base
[02:21:57] (native) install qemu-user-static-repack qemu-user-static-repack-binfmt
[02:21:58] Register qemu binfmt (aarch64)
[02:21:58] (buildroot_aarch64) install alpine-base
/ # echo $PATH
/usr/lib/ccache/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
/ # su pmos -c 'echo $PATH'
/usr/lib/ccache/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
/ # apk add shadow
(1/2) Installing linux-pam (1.3.0-r0)
(2/2) Installing shadow (4.5-r0)
Executing busybox-1.27.2-r7.trigger
OK: 9 MiB in 18 packages
/ # su pmos -c 'echo $PATH'
/bin:/usr/bin
/ #
```

**Comment by ollieparanoid** [comment]
`shadow` provides its own `su` binary. When it's not installed, the version from BusyBox gets used.

After some trying, I've found that both `su` versions print the correct path when using the `-l` flag (`-` prints the wrong `PATH` with the `shadow` version, and `--login` is not defined in the BusyBox version).

```
/ # su --help
BusyBox v1.27.2 (2017-12-12 10:38:25 GMT) multi-call binary.

Usage: su [-lmp] [-] [-s SH] [USER [SCRIPT ARGS / -c 'CMD' ARG0 ARGS]]

Run shell under USER (by default, root)

        -,-l    Clear environment, go to home dir, run shell as login shell
        -p,-m   Do not set new $HOME, $SHELL, $USER, $LOGNAME
        -c CMD  Command to pass to 'sh -c'
        -s SH   Shell to use instead of user's default
/ # su pmos -l -c 'echo $PATH'
/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
/ # apk add shadow
(1/2) Installing linux-pam (1.3.0-r0)
(2/2) Installing shadow (4.5-r0)
Executing busybox-1.27.2-r7.trigger
OK: 9 MiB in 18 packages
/ # su --help
Usage: su [options] [LOGIN]

Options:
  -c, --command COMMAND         pass COMMAND to the invoked shell
  -h, --help                    display this help message and exit
  -, -l, --login                make the shell a login shell
  -m, -p,
  --preserve-environment        do not reset environment variables, and
                                keep the same shell
  -s, --shell SHELL             use SHELL instead of the default in passwd

/ # su pmos -l -c 'echo $PATH'
/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```

I will prepare a PR that always uses the `-l` option.

**Comment by drebrez** [comment]
@ollieparanoid nice that you figured out what the problem was. For me is fine with both solution, as you prefer 😉

**Comment by ollieparanoid** [comment]
Closed in #1085.
